cmd/bot/run.go

```diff
@@ -76,65 +76,79 @@ func runCommand(ctx *cli.Context) error {
+					return
 				}
+			} else if strings.Contains(err.Error(), "L2OutputOracle: cannot get output for a block that has not been proposed") {
+				// Since the unproven withdrawals are sorted by the on-chain order, we can break here because we know
+				// that the subsequent of the withdrawals are not ready to be proven yet.
+				return
+			} else {
+				// TODO handle other errors
+				log.Error("FinalizeMessage", "error", err.Error())
 			}
 		}
+	}
+}
 
-		unfinalizeds := make([]core.L2ContractEvent, 0)
-		if result := db.Order("block_time asc").Where("proven = true AND finalized = false").Limit(limit).Find(&unfinalizeds); result.Error != nil {
-			log.Error("querying l2_contract_events", "error", result.Error)
-			continue
-		} else if len(unfinalizeds) > 0 {
-			for _, unfinalized := range unfinalizeds {
-				if unfinalized.BlockTime+cfg.Misc.ChallengeTimeWindow < time.Now().Unix() {
-					err := processor.FinalizeMessage(&unfinalized)
-					if err != nil {
-						if strings.Contains(err.Error(), "OptimismPortal: withdrawal has already been finalized") {
-							// The withdrawal has already finalized, mark it
-							unfinalized.Finalized = true
-							result := db.Save(&unfinalized)
-							if result.Error != nil {
-								log.Crit("update finalized l2_contract_events", "error", result.Error)
-							}
-						} else if strings.Contains(err.Error(), "OptimismPortal: withdrawal has not been proven yet") || strings.Contains(err.Error(), "OptimismPortal: proven withdrawal finalization period has not elapsed") {
-							break
-						} else {
-							log.Crit("FinalizeMessage", "error", err.Error())
-						}
-					}
+func ProcessUnfinalizedBotDelegatedWithdrawals(ctx context.Context, log log.Logger, db *gorm.DB, l1Client *core.ClientExt, l2Client *core.ClientExt, cfg core.Config) {
+	processor := core.NewProcessor(log, l1Client, l2Client, cfg)
+	limit := 1000
+	maxBlockTime := time.Now().Unix() - cfg.Misc.ChallengeTimeWindow
+
+	unfinalizeds := make([]core.L2ContractEvent, 0)
+	result := db.Order("block_time asc").Where("proven = true AND finalized = false AND block_time < ?", maxBlockTime).Limit(limit).Find(&unfinalizeds)
+	if result.Error != nil {
+		log.Error("failed to query l2_contract_events", "error", result.Error)
+		return
+	}
+
+	for _, unfinalized := range unfinalizeds {
+		err := processor.FinalizeMessage(ctx, &unfinalized)
+		if err != nil {
+			if strings.Contains(err.Error(), "OptimismPortal: withdrawal has already been finalized") {
+				// The withdrawal has already finalized, mark it
+				result := db.Model(&unfinalized).Update("finalized", true)
+				if result.Error != nil {
+					log.Error("failed to update finalized l2_contract_events", "error", result.Error)
+					return
 				}
+			} else if strings.Contains(err.Error(), "OptimismPortal: withdrawal has not been proven yet") || strings.Contains(err.Error(), "OptimismPortal: proven withdrawal finalization period has not elapsed") {
```

Comment:
`OptimismPortal: withdrawal has not been proven yet` appeared, but all the data we queried before were `proven = true`, which indicates that there is a problem with the indexer. I think there should be an error log here.

Reply:
Make sense.
Updated at [6df5778](https://github.com/bnb-chain/opbnb-bridge-bot/pull/7/commits/6df57789b0162300332975e491c9434313f6beb9), @welkin22 PTAL